core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -70,20 +76,66 @@ object ConsumerGroupCommand {
     }
   }
 
-  def list(zkUtils: ZkUtils) {
-    zkUtils.getConsumerGroups().foreach(println)
+  def list(zkUtils: ZkUtils, opts: ConsumerGroupCommandOptions) {
+    val useNewConsumer = opts.options.has(opts.newConsumerOpt)
+    if (!useNewConsumer)
+      zkUtils.getConsumerGroups().foreach(println)
+    else {
+      val adminClient = createAndGetAdminClient(opts)
+      adminClient.listAllGroupsFlattened().foreach(x => println(x.groupId))
+    }
+  }
+
+  def createAndGetAdminClient(opts: ConsumerGroupCommandOptions): AdminClient = {
+    AdminClient.createSimplePlaintext(opts.options.valueOf(opts.bootstrapServerOpt))
   }
 
   def describe(zkUtils: ZkUtils, opts: ConsumerGroupCommandOptions) {
+    val useNewConsumer = opts.options.has(opts.newConsumerOpt)
+    val group = opts.options.valueOf(opts.groupOpt)
     val configs = parseConfigs(opts)
     val channelSocketTimeoutMs = configs.getProperty("channelSocketTimeoutMs", "600").toInt
     val channelRetryBackoffMs = configs.getProperty("channelRetryBackoffMsOpt", "300").toInt
-    val group = opts.options.valueOf(opts.groupOpt)
-    val topics = zkUtils.getTopicsByConsumerGroup(group)
-    if (topics.isEmpty) {
+    def warnNoTopicsForGroupFound: Unit = {
       println("No topic available for consumer group provided")
     }
-    topics.foreach(topic => describeTopic(zkUtils, group, topic, channelSocketTimeoutMs, channelRetryBackoffMs))
+
+    println("%s, %s, %s, %s, %s, %s, %s"
+      .format("GROUP", "TOPIC", "PARTITION", "CURRENT OFFSET", "LOG END OFFSET", "LAG", "OWNER"))
+
+    if (!useNewConsumer) {
+      val topics = zkUtils.getTopicsByConsumerGroup(group)
+      if (topics.isEmpty) {
+        warnNoTopicsForGroupFound
+      }
+      topics.foreach(topic => describeTopic(zkUtils, group, topic, channelSocketTimeoutMs, channelRetryBackoffMs, opts))
+    } else {
+      val groupSummary: GroupSummary = createAndGetAdminClient(opts).describeGroup(group)
+      if (groupSummary.protocolType != ConsumerProtocol.PROTOCOL_TYPE)
+        println(s"Group ${group} is not a consumer group")
+
+      val members: List[MemberSummary] = groupSummary.members
+      val owners: Map[TopicPartition, String] = members.flatMap {
+        case member =>
+          val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(member.assignment))
```

Comment:
Would be nice to keep this kind of logic encapsulated in AdminClient. Maybe we can change the current describeConsumerGroup so that it returns (GroupSummary, Map[String, List[TopicPartition]])?